backend/user/views.py

```diff
@@ -109,27 +98,37 @@ class NotificationAlertView(APIView):
     permission_classes = [B2BPermission("urn:pennlabs:*") | IsAuthenticated]
 
     def post(self, request):
-        users = (
+        usernames = (
             [self.request.user.username]
             if request.user and request.user.is_authenticated
             else request.data.get("users", list())
         )
+
         service = request.data.get("service")
         title = request.data.get("title")
         body = request.data.get("body")
         delay = max(request.data.get("delay", 0), 0)
-        is_dev = request.data.get("is_dev", False)
+        is_dev = request.data.get("is_dev", "false").lower() == "true"
 
         if None in [service, title, body]:
             return Response({"detail": "Missing required parameters."}, status=400)
-        if service not in dict(NotificationSetting.SERVICE_OPTIONS):
+
+        if not (service_obj := NotificationService.objects.filter(name=service).first()):
             return Response({"detail": "Invalid service."}, status=400)
+        
+        users_with_service = service_obj.enabled_users.filter(username__in=usernames)
+
+        tokens = list(IOSNotificationToken.objects.filter(user__in=users_with_service, is_dev=is_dev).values_list(
```

Comment:
just an opinion but probably prettier to extract this into a `get_tokens()` method within the `post` method but I defer to your judgement